frontend/src/visualization/filtering/Filter.tsx

```diff
@@ -64,140 +70,75 @@ function TabPanel(props: TabPanelProps) {
   );
 }
 
-/**
- * Fetches nodeTypes from the schemaService.
- *
- * @param schemaService - the schemaService the data is fetched from
- * @param cancellation - the cancellation token
- */
-function fetchNodeTypes(
-  schemaService: SchemaService,
-  cancellation: CancellationToken
-): Promise<NodeType[]> {
-  return schemaService.getNodeTypes(cancellation);
-}
-
-/**
- * Fetches edgeTypes from the schemaService.
- *
- * @param schemaService - the schemaService the data is fetched from
- * @param cancellation - the cancellation token
- */
-function fetchEdgeTypes(
-  schemaService: SchemaService,
-  cancellation: CancellationToken
-): Promise<EdgeType[]> {
-  return schemaService.getEdgeTypes(cancellation);
-}
-
 const Filter = (): JSX.Element => {
   // hooks
   const classes = useStyles();
   const [tabIndex, setTabIndex] = React.useState(0);
   const [open, setOpen] = React.useState(false);
 
-  const schemaService = useService(SchemaService, null);
-
-  const filterStore = useService(FilterQueryStore);
-
-  const nodeConditionsRef = useRef<(FilterCondition | null)[]>([]);
-  const edgeConditionsRef = useRef<(FilterCondition | null)[]>([]);
-
-  function updateQuery() {
-    const nodeConditions = nodeConditionsRef.current.filter(
-      (condition) => condition !== null
-    ) as FilterCondition[];
-
-    const edgeConditions = edgeConditionsRef.current.filter(
-      (condition) => condition !== null
-    ) as FilterCondition[];
+  const filterStateStore = useService<FilterStateStore>(FilterStateStore);
```

Comment:
TS can Inference the type automatically.
```suggestion
  const filterStateStore = useService(FilterStateStore);
```